### PR TITLE
4444 make it impossible to change the metric type if that would mean sources have to be deleted

### LINKED ISSUE #4444: When changing the metric type, don't delete sources that do not support the new type
**Is your feature request related to a problem? Please describe.**
Changing the type of a metric results in sources that do not support the new type to be deleted. If a user changes the type by accident and changes the type back, this may result in loss of configured sources.

**Describe the solution you'd like**
Don't remove sources that don't support the metric type, but keep them. Flag in the UI that a source is ignored (with a grey instead of red background label) because it doesn't support the metric. Ignore the source in the collector (that should happen automatically as there will be no collector the metric/source combination).

**Describe alternatives you've considered**
- Warning the user with a dialog. However, it's easy to ignore dialogs and Quality-time doesn't use dialogs in other place either.
- Making it impossible to change the type, but that also blocks the user when they know what they're doing.

**Additional context**
Related bug: #4443. It's probably easier to implement this if the bug is fixed first.

### PR DESCRIPTION
**Pull request by fniessink**
Closes #4444.

### CODE REVIEW

File: components/external_server/src/routes/metric.py [hunk 1]
```diff
@@ -132,12 +132,8 @@ def post_metric_attribute(metric_uuid: MetricId, metric_attribute: str, database
...
     metric[metric_attribute] = new_value
     if metric_attribute == "type":
-        sources_to_keep = {
-            key: source
-            for key, source in metric["sources"].items()
-            if source["type"] in data_model["metrics"][new_value]["sources"]
-        }
-        metric.update(default_metric_attributes(database, new_value), sources=sources_to_keep)
+        # Update the metric attributes, but keep the sources
+        metric.update(default_metric_attributes(database, new_value), sources=metric["sources"])
```
Comment (wkoot) [inline components/external_server/src/routes/metric.py:136]: This is simply returning to the original right?

File: components/frontend/src/measurement/MeasurementSources.test.js [hunk 2]
```diff
@@ -3,12 +3,15 @@ import { render, screen } from '@testing-library/react';
...
+
 it('renders one measurement source', () => {
     render(
-        <DataModel.Provider value={{}}>
+        <DataModel.Provider value={dataModel}>
             <MeasurementSources metric={
                 {
-                    sources: { source_uuid: { name: "Source name" } },
+                    type: "metric_type",
+                    sources: { source_uuid: { type: "sourcr_type", name: "Source name" } },
```
Comment (wkoot) [inline components/frontend/src/measurement/MeasurementSources.test.js]: looks like a typo `sourcr_type`

File: components/frontend/src/source/Source.js [hunk 3]
```diff
@@ -88,21 +89,47 @@ function Parameters({ metric, source, source_uuid, connection_error, parse_error
...
     const parseError = measurement_source?.parse_error || "";
-    const configurationTabLabel = connectionError || parseError ? <Label color='red'>{"Configuration"}</Label> : "Configuration";
+    const referenceManualURL = `https://quality-time.readthedocs.io/en/v${process.env.REACT_APP_VERSION}/reference.html`;
+    const configErrorMessage = <>
+        <p>
+            {sourceName} cannot be used to measure {metricName}. This configuration error occurs if the type of a
+            metric is changed to a metric type that is not supported by the configured source type(s).
+        </p>
+        <p>
+            There are several ways to remediate this error:
```
Comment (wkoot) [inline components/frontend/src/source/Source.js]: Perhaps use "remedy" as it is also correct, but simpler

File: components/shared_server_code/src/shared/model/measurement.py [hunk 4]
```diff
@@ -261,8 +262,12 @@ def update_measurement(self) -> None:
             self[scale].update_value_and_status()
 
     def sources_ok(self) -> bool:
-        """Return whether there are sources and none have parse or connection errors."""
+        """Return whether there are sources and none have configuration, parse, or connection errors."""
         sources = self.sources()
+        for source in sources:
```
Comment (wkoot) [inline components/shared_server_code/src/shared/model/measurement.py:267]: Does it make sense to short-circuit parse and connection errors before looping over the sources?

File: docs/src/changelog.md [hunk 5]
```diff
@@ -8,6 +8,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
...
 
+## [Unreleased]
+
+### Deployment notes
+
+If your currently installed *Quality-time* version is not v4.4.0, please read the v4.4.0 deployment notes.
+
+### Fixed
+
+- When changing the type of a metric with configured sources, retain the sources whether they support the new metric type or not. Flag the sources that do not support the new metric type as a 'configuration error' and explain how the user can remediate the situation. Fixes [#4443](https://github.com/ICTU/quality-time/issues/4443) and closes [#4444](https://github.com/ICTU/quality-time/issues/4444).
```
Comment (wkoot) [inline docs/src/changelog.md]: If earlier text was changed to remedy, then this one should be as well

File: components/external_server/src/routes/metric.py (hunk 1 above)
Comment (fniessink) [reply components/external_server/src/routes/metric.py:136]: It sets default attribute values for the new type, yes. Except that the sources are not removed.

File: components/frontend/src/source/Source.js (hunk 3 above)
Comment (fniessink) [reply components/frontend/src/source/Source.js]: Or "fix", even easier :-)

File: components/shared_server_code/src/shared/model/measurement.py (hunk 4 above)
Comment (fniessink) [reply components/shared_server_code/src/shared/model/measurement.py:267]: I don't think it is worth it, metrics typically have only one or a handful of sources.

**Pull request merged**